### ISSUE TITLE
Adding `tbl-colwidths` as a label at the start of Python code cells no longer sets the column width of Markdown tables

### ISSUE THREAD

**Issue by carstrb** [quarto-dev/quarto-cli#9200]
### Bug description

In `v1.3.450` of quarto, Using labels at the top of code cells to specify table attributes such as `tbl-colwidths` allowed for per-table formatting when generating a PDF. However, the latest version of quarto (`v1.4.551`) seems to no longer to adhere to this command, and what results is a table that overruns the margins/boundary of the page that it is rendered on.

### Steps to reproduce

The document below renders a simple table using data from `mt-cars`

````qmd
---
format: pdf
echo: false
---

```{python}
#| tbl-colwidths: [20,10,15,10,15,15,15,15,15,15,15,10]

import pandas as pd
from IPython.display import Markdown
from tabulate import tabulate

df = pd.read_csv('https://gist.githubusercontent.com/ZeccaLehn/4e06d2575eb9589dbe8c365d61cb056c/raw/64f1660f38ef523b2a1a13be77b002b98665cdfe/mtcars.csv')

df.columns = [
    "Car Name",
    "Fuel Cons. (mpg)",
    "# Cylinders",
    "Displ. (cu.in.)",
    "Gross HP",
    "Rear axle ratio",
    "Weight (1000 lbs)",
    "1/4 mile time",
    "Engine",
    "Trans.",
    "# forward gears",
    "# carburetors",
]

Markdown(
    tabulate(
        df,
        headers="keys",
        showindex=False,
    )
)
```
````

To run, you'll need to copy the above contents into a `.qmd` file, and ensure the Python packages below have been installed. The command I use to render is `quarto render <DOCUMENT_NAME.qmd>`

The packages required to render this document as a pdf are as follows:

```
jupyter==1.0.0
pandas==2.2.1
ipython==8.22.2
tabulate==0.9.0
```

### Expected behavior

A table that has been given `tbl-colwidths` arguments should remain within the margins of the page, with multi-line/wrapped headers based on the specified width of its respective column. See the following output below:

![Screenshot 2024-03-26 144725](https://github.com/quarto-dev/quarto-cli/assets/150830409/899c4631-ce58-4a69-854c-7accd02d918b)

### Actual behavior

In the case of `quarto: v1.4.551`, there is no formatting applied to the columns of the table, and so the result is a table that spills outside of the page right margin:
![Screenshot 2024-03-26 145255](https://github.com/quarto-dev/quarto-cli/assets/150830409/6a1eb5c0-4a4b-423b-8b31-8eaaa7a655b6)

### Your environment

- IDE: VSCode [WSL: Ubuntu]: 1.87.2

- Base OS: Windows 11 Home
- Version:  23H2
- OS Build: 22631.3296

- WSL Distribution: Ubuntu 22.04.4 LTS
- Release: 22.04
- Codename: jammy

### Quarto check output

```
Quarto 1.4.551
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.551
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.03.13
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/carson/.TinyTeX/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2
      Path: /home/carson/Code/Playground/table_demo/quarto-table-demo/.venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Comment by carstrb** [comment]
I should mention also that the most recent version of `quarto` where this issue is not present, is `v.1.3.450`

**Comment by cscheid** [comment]
I can repro this on `main`: the `tbl-colwidth` field is not getting forwarded to the table element being produced by the Jupyter cell.

**Comment by carstrb** [comment]
I can also reproduce this using `v1.5.27` (pre-release) of `quarto`:

### Quarto check output

```
Quarto 1.5.27
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.27
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.03.13
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/carson-oceansync/.TinyTeX/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2
      Path: /home/carson-oceansync/OceanSync/Playground/table_demo/quarto-table-demo/.venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

### Output

![Screenshot 2024-03-27 125225](https://github.com/quarto-dev/quarto-cli/assets/150830409/ba6e30b7-1052-45ae-92c7-7f52e70b9f9d)

**Comment by cscheid** [comment]
The specific bug is that a non-crossreferenceable table element is not getting its elements properly forwarded anymore. I'll fix this shortly. In the meantime, you might be able to work around it by adding a label and caption to your code cell, like this:

````
```{python}
#| tbl-colwidths: [20,10,15,10,15,15,15,15,15,15,15,10]
#| label: tbl-1
#| tbl-cap: A caption

import pandas as pd
from IPython.display import Markdown
...
```
````

Which then produces this output:

<img width="248" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/3b6424c9-34ea-417a-a947-bb3f7648241f">

**Comment by cscheid** [comment]
This is now fixed on `main`. We need a separate backport because the way we fixed it going forward in 1.5 introduces a new concept, and we don't want to do that in a patch release.

**Comment by cscheid** [comment]
(reopening for 1.4 fix)

**Comment by cscheid** [comment]
Fixed on `v1.4`.
